quickwit/quickwit-rest-client/src/rest_client.rs

```diff
@@ -149,17 +152,30 @@ impl QuickwitClient {
             }
             IngestSource::Stdin => Box::new(tokio::io::stdin()),
         };
-        let mut buf_reader = BufReader::new(reader);
+        let buf_reader = BufReader::new(reader);
+        let peekable_reader = LinesStream::new(buf_reader.lines()).peekable();
+        pin_mut!(peekable_reader);
 
         let start = Instant::now();
         let mut num_bytes_sent = 0;
 
         loop {
             let mut buffer = Vec::with_capacity(INGEST_CHUNK_SIZE_IN_BYTES);
-            for _ in 0..5_000 {
-                if buf_reader.read_until(b'\n', &mut buffer).await? == 0 {
-                    break;
-                }
+            while let Some(read_result) = peekable_reader
```

Comment:
Indeed, this is not very readable. I prefer the "simple" solution, which we can improve to be accurate. We have to pop the last line from the current buffer and push it to a new one.

Reply:
Yeah, I thought of this but I felt that's even more complicated and less idiomatic than the peelable solution. 
Please see the changes.